lib/BackgroundJob/RetentionJob.php

```diff
@@ -219,7 +234,14 @@ private function checkFileId(int $fileId): Node {
 			throw new NotFoundException('No node for file ' . $fileId . ' and user ' . $userId);
 		}
 
-		return array_shift($nodes);
+		foreach ($nodes as $node) {
+			if ($node->getPermissions() & \OCP\Constants::PERMISSION_DELETE) {
```

Comment:
I would use `$node->isDeletable()` instead for simplicity (unless there is any case in which they give different results, which I do not know).